File: context_tree_cli/rules/framework.py
"""Rule: check .context-tree/ framework presence."""

from __future__ import annotations

from context_tree_cli.repo import Repo

SEED_TREE_URL = "https://github.com/agent-team-foundation/seed-tree"


def evaluate(repo: Repo) -> dict:
    tasks = []
    if not repo.has_framework():
        tasks.append(
            f"`.context-tree/` not found — run `context-tree init` to clone the"
            f" framework from {SEED_TREE_URL}"
        )
    return {"group": "Framework", "order": 1, "tasks": tasks}

File: .claude/skills/first-tree-cli-framework/references/repo-snapshot/src/rules/framework.ts
import type { Repo } from "#src/repo.js";
import type { RuleResult } from "#src/rules/index.js";
import { SKILL_ROOT } from "#src/runtime/asset-loader.js";

const FIRST_TREE_REPO_URL = "https://github.com/agent-team-foundation/first-tree";

export function evaluate(repo: Repo): RuleResult {
  const tasks: string[] = [];
  if (!repo.hasFramework()) {
    tasks.push(
      `\`${SKILL_ROOT}/\` not found — run \`context-tree init\` to install the framework skill from ${FIRST_TREE_REPO_URL}`,
    );
  }
  return { group: "Framework", order: 1, tasks };
}
